app.py

```diff
@@ -496,13 +499,141 @@ def update_public_chart(start_date, end_date):
     return figure
 
 
+"""Query Page Callbacks"""
+
+@callback(
+    Output("total-games-value-query", "children"),
+    Input("date-picker-query", "start_date"),
+    Input("date-picker-query", "end_date"),
+    prevent_initial_call=True,
+)
+def update_total_games_value(start_date, end_date):
+    """
+    Updates the total-games-value card displayed on the dashboard.
+
+    When the date range is changed, this function fetches the total count of games played in the given date range from the database,
+    formats it to have comma separators, and returns it to be displayed on the dashboard.
+
+    Args:
+        start_date (str): The start date of the selected date range.
+        end_date (str): The end date of the selected date range.
+
+    Returns:
+        str: The count of total games formatted to have comma separators.
+    """
+    # Validate that both start_date and end_date are not None.
+    if start_date is None or end_date is None:
+        raise PreventUpdate
+
+    # Fetch the total count of games played in the given date range from the database.
+    mariadb_connection = connect_to_mariadb()
+    cursor = mariadb_connection.cursor()
+    target_table = get_target_table()
+    cursor.execute(
+        f"SELECT COUNT(*) FROM {target_table} WHERE START_TIME BETWEEN ? AND ?",
+        (start_date, end_date),
+    )
+    logging.debug(f"Fetched the count of total games from {target_table} from database")
+    games_count = cursor.fetchone()[0]
+    cursor.close()
+    mariadb_connection.close()
+
+    return f"{games_count:,}"
+
+
+@callback(
+    Output("table", "data"),
+    Output("constraints-modal", "is_open"),
+    Input("total-games-value-query", "children"),
+    State('date-picker-query', 'start_date'),
+    State('date-picker-query', 'end_date'),
+    prevent_initial_call=True
+)
+def update_table(total_games, start_date, end_date):
+    """
+    Update the table with data from the database based on the selected date range.
+
+    Args:
+        start_date (str): The start date of the selected date range.
+        end_date (str): The end date of the selected date range.
+
+    Returns:
+        list[dict]: A list of dictionaries containing the data to be displayed in the table.
+    """
+    # Validate that both start_date and end_date are not None.
+    if start_date is None or end_date is None:
+        raise PreventUpdate
+    
+    # Inform the user that the query cannot be processed because the size of the data to process exceeds limitations.
+    if int(total_games.replace(",", "")) > 5000:
```

Comment:
What is the replace for?

The limit should be configurable.

Reply:
> What is the replace for?

I get an exception when applying `int()` to convert `total_games` from a string to an integer because of the existence of comma characters in the string when it represents numbers greater than 1,000. The `replace()` gets rid of any commas in the `total_games` before it is converted to an integer.

Reply:
Is this using the string shown on the page? Why is this not using the actual number?

A random replace that works with one specific number formatting of some locale is not great.

Reply:
Using the string. I didn't know that this wasn't great. I'll see if it is possible to use the actual number.